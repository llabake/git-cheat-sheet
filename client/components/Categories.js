import React from 'react';
import { connect } from 'react-redux';

import Category from './Category';

export default connect(state => ({
  categories: state.categories,
  searchResults: state.search,
}))(({ categories, searchResults }) => {
  const data = searchResults || categories;
  return (
    data.map(category => (
      <Category key={category._id} category={category} />
    ))
  );
});
