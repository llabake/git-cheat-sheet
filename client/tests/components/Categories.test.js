import React from 'react';
import { shallow } from 'enzyme';
import Categories from '../../components/Categories';

describe("<Categories />", () => {
  const props = {
    categories: [],
    searchResults: null,
  };
  it("renders without crashing", () => {
    const wrapper = shallow(<Categories {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});