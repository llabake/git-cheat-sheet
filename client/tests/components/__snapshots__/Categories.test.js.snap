// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Categories /> renders without crashing 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
